parsec/serde/schema.py

```diff
@@ -144,7 +155,7 @@ def _dump(self, obj, update_fields=True, **kwargs):
             return MarshalResult(None, {"_schema": "Unsupported object type: %s" % obj_type})
 
         result = schema.dump(obj, many=False, update_fields=update_fields, **kwargs)
-        if result.data:
+        if result.data and self.type_field not in result.data:
```

Comment:
Didn't we agree to remove this part altogether ? (considering it's the schema responsibility to do the serialization, so doing `result.data[self.type_field] = obj_type` after it is a bad idea)

We should instead have a `assert self.type_field in result.data` check (another equivalent safety check would be to ensure all the schema defined in `self._instantiated_schemas` define a `CheckedEnumConstant` field named after `self.type_field`, this is a bit harder to implement but s0 much Rox0r for sP33D !)

Reply:
Now schema are inspected when `_instanciated_schemas` is created:
```python
        for k, v in self.type_schemas.items():
            schema_instance = v if isinstance(v, Schema) else v()
            if self.type_field not in schema_instance.fields:
                raise ValueError(f"{schema_instance} needs to define '{self.type_field}' field")
```